lib/config/config.js

```diff
@@ -135,30 +123,18 @@ function createConfig(opts, logger) {
 
 class Config {
   constructor(opts, logger) {
-    this.disableMetricsRegExp = [];
-    this.ignoreUrlStr = [];
-    this.ignoreUrlRegExp = [];
-    this.ignoreUserAgentStr = [];
-    this.ignoreUserAgentRegExp = [];
-    this.elasticsearchCaptureBodyUrlsRegExp = [];
-    this.transactionIgnoreUrlRegExp = [];
-    this.sanitizeFieldNamesRegExp = [];
-    this.ignoreMessageQueuesRegExp = [];
+    // TODO: remove this call once config and logging-preamble tests are refactored
+    readEnvOptions();
 
     const isLambda = isLambdaExecutionEnvironment();
     const envOptions = getEnvironmentOptions();
 
-    // If we didn't find a config file on process boot, but a path to one is
-    // provided as a config option, let's instead try to load that
-    if (confFileOpts === null && opts && opts.configFile) {
-      confFilePath = path.resolve(opts.configFile);
-      confFileOpts = getFileOptions(opts.configFile);
-    }
+    setStartOptions(opts);
```

Comment:
Note for reviewer: now `setStartOptions` handles if a new config file is set during agent start so no need for the previous code.